fulltext/__init__.py

```diff
@@ -71,6 +72,32 @@ def _import_backends():
     LOGGER.info('Loaded backends: %s', ', '.join(BACKENDS.keys()))
 
 
+def is_binary(f):
+    """Return True if binary mode."""
+    # Python 2 makes no distinction.
+    if not PY3:
+        return True
+
+    # If the file has a mode, and it contains b, it is binary.
+    if 'b' in getattr(f, 'mode', ''):
+        return True
+
+    # If it has a decoding attribute with a value, it is text mode.
+    if getattr(f, "encoding", None):
+        return False
+
+    # Can we sniff?
+    try:
+        f.seek(0, os.SEEK_CUR)
```

Comment:
will this work on Windows? (actually not sure if this is supposed to work on win)
also consider adding a test which makes sure the offset of a file-like obj is correctly reset to its initial position

Reply:
We do want this to work on Windows. I don't see any notes in Python docs about this not working on Windows.